modules/tracking/src/trackerKCF.cpp

```diff
@@ -486,23 +570,23 @@ namespace cv{
     if(patch.rows==0 || patch.cols==0)return false;
 
     // extract the desired descriptors
-    switch(params.descriptor){
-      case GRAY:
-        if(img.channels()>1)cvtColor(patch,patch, CV_BGR2GRAY);
-        patch.convertTo(patch,CV_64F);
-        patch=patch/255.0-0.5; // normalize to range -0.5 .. 0.5
+    switch(desc){
+      default:
```

Comment:
yes the default descriptor will be GRAY, i also surprised that this code working, ok i will rewrite in standard form with default in the last to avoid any unwanted results.